pkg/processor/test/suite/suite.go

```diff
@@ -239,6 +239,21 @@ func (suite *TestSuite) DeployFunctionAndRedeploy(createFunctionOptions *platfor
 	suite.deployFunction(createFunctionOptions, onAfterSecondContainerRun, false)
 }
 
+func (suite *TestSuite) DeployFunctionExpectErrorAndRedeploy(createFunctionOptions *platform.CreateFunctionOptions,
```

Comment:
This seems like a very "contrived" utility method. Isn't it easier to explicitly do that in the few tests that use this special flow?
Where does it end? `DeployfunctionExpectErrorAndRedeployAndExpectSuccessAndRedployTwiceMoreAndExpectErrors` ?

Reply:
It is actually very convenient and useful. it promises cleanup and do the heavy lifting of deploying for you.
the alternative is having a callback hell.

Reply:
That is not currect. You don't need this to promise Cleanup at all.
What you mean to say, I think is rhat the current callback system means multi deployment flows will give you a nested callback structure like a context manager would. That's how they are designed.
While I'm not a huge fan of that, that is what the callback mechanism here gives you and this is how it works. And it dictates that if you want to chain them it comes out nested. Here this helper takes the nested system and tried to pretend it's flat for specific chain of events.
Exactly like my first comment stated, that is contrived - it's not getting rid of the nesting problem or approach and it's not going with it, but half hiding it, which I think is worse and will yield a wide variety of such flow combinations and helpers and inconsistent look and feel (which is already the case ) to the tests
Given that you didn't